typing/typecore.ml

```diff
@@ -984,6 +995,11 @@ module Constructor = NameChoice (struct
   let get_type cstr = cstr.cstr_res
   let lookup_all_from_type loc usage path env =
     Env.lookup_all_constructors_from_type ~loc usage path env
+
+  let extra_spellchecking_candidates filter env =
+    let add_valid x acc = if filter x then get_name x::acc else acc in
+    Env.fold_constructors add_valid None env []
```

Comment:
There are two functions here that look pretty similar from the outside: Env.lookup_all_constructors_from_type` and the `Env.fold_constructors` plus same-path filtering that is used in practice by the callers of `extra_spellcheck_candidates`. 

I wonder if we could have just one function that does both these things, and whether that would suffice to get parts of these PRs that now have to explicitly call `extra_spellchecking_candidates` to behave "as they should" (and do with the PR).

@Octachron said (in an audio-conference-review) that the first is more efficient in the non-extension case. I wonder if this would be a problem in practice, given the way these functions are used.